### DISCUSSION TITLE
Using getServerSideProps gives Gateway Timeout Error in production server and works fine in local server.

### DISCUSSION THREAD

**Discussion by JwngmaBasumatary** [vercel/next.js#45734] [Help]
So This is my getServerSideProps Code : 

```
export async function getServerSideProps(context) {
  const { vendorId } = context.params;
  const res = await fetch(
    `${process.env.NEXT_PUBLIC_API}/api/v1/user/vendor-details?vendorId=${vendorId}&page=1&limit=8`,
    {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  const data = await res.json();

  return {
    props: {
      vendorDetails: data.data.vendor,
      vendorProducts: data.data.products,
      storeId: vendorId,
    },
  };
}
```

This Code works well in the local server but when i deploy it on vercel and i open this page it shows gateway timeout. I have Checked the environment variables as well but still i get the timeout issue. What could be the issue?

**Comment by icyJoseph** [comment]
Well assuming that `process.env.NEXT_PUBLIC_API` is not the same server. I've written so much againts this, that I'll skip it for this one, if you are curious read this part of the docs: https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#getserversideprops-or-api-routes

I'd say that it might be a good idea to try and debug this with a try/catch. 

Whenever you do async work it is a good idea to handle exceptions. In your case, we need to make sure that fetch is working correctly, and then we need to make sure that that `.json()` also works correctly.

```js
export async function getServerSideProps(context) {
    try {
        const { vendorId } = context.params;
        const res = await fetch(`${process.env.NEXT_PUBLIC_API}/api/v1/user/vendor-details?vendorId=${vendorId}&page=1&limit=8`, {
            method: "GET",
            headers: {
                "Content-Type": "application/json",
            },
        });
        const data = await res.json();

        return {
            props: {
                vendorDetails: data.data.vendor,
                vendorProducts: data.data.products,
                storeId: vendorId,
            },
        };
    } catch (reason) {
        console.log("API:", process.env.NEXT_PUBLIC_API);
        console.log("vendorId:", context.params.vendorId);

        // just log it out to stdout
        if (reason instanceof Error) {
            console.log(reason.message);
        } else {
            console.log("Unexpected exception", reason);
        }
       
        return {
            props: {
                /*  some sensible default props in case of error */
            },
        };
    }
}
```

Collect some logs and let's see what you get.

**Reply by JwngmaBasumatary** [reply]
Hi, Thanks for the help. 

The Logs are empty. 

I actually used logs before to check if some error occurs while fetching but not a single log command inside the getServerSideProps is shown in vercel. So my guess is this error occurs outside before getServerSide is called because Vercel doesn't show me any logs, only this : 
![image](https://user-images.githubusercontent.com/97206612/217794833-e76ffede-e66f-499d-9011-ef6153732e7d.png)
